File: exercises/pi/pi.cpp
#include <cstdio>
#include <math.h>

int main()
{
	const int steps = 1000000000;
	const double dx = 1./steps;
    const double pi = std::acos(-1.0);

    double sum = 0.0;

    for(int i=0; i<steps; i++)
    {
        double x = (i-0.5) * dx;
        sum += 4.0 / (1.0 + x*x);
    }

    sum *= dx;

	printf("%.15f %.15e\n", sum, abs(pi-sum));

	return 0;
}

File: exercises/pi/pi.f90
program pi_serial

  implicit none


  integer, parameter::steps=1000000000
  integer i
  double precision dx,x,sum,pi
  
  pi=acos(-1.d0)
  dx=1.d0/dble(steps)
  sum=0.d0

  do i=1,steps
     x=(dble(i)-.5d0)*dx
     sum=sum+4.d0/(1.d0+x*x)
  enddo

  sum=sum*dx

  print *,sum,abs(pi-sum)/pi

end program pi_serial
